recipes/wavelet_buffer/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+
+    def requirements(self):
+        self.requires("openblas/0.3.20")
+        self.requires("blaze/3.8")
+        self.requires("libjpeg-turbo/2.1.2")
+        self.requires("cimg/3.0.2")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        get(
+            self,
+            **self.conan_data["sources"][self.version],
+            destination=self.source_folder,
+            strip_root=True,
+        )
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CONAN_EXPORTED"] = True
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 20
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "8",
+            "clang": "13",
+            "apple-clang": "13",
+        }
+
+    def validate(self):
+        if self.options.shared:
+            raise ConanInvalidConfiguration(
+                "{} requires static linking".format(self.name)
+            )
+
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
+
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn(
+                "{} recipe lacks information about the {} compiler support.".format(
+                    self.name, self.settings.compiler
+                )
+            )
+        else:
+            if Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                        self.name,
+                        self._minimum_cpp_standard,
+                        self.settings.compiler,
+                        self.settings.compiler.version,
+                    )
+                )
+
+        cimg = self.dependencies["cimg"]
+        if cimg.options.enable_fftw:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires the option 'cimg:enable_fftw=False'"
+            )
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(
+            self,
+            pattern="LICENSE",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["wavelet_buffer", "sf_compressor"]
```

Comment:
```suggestion
        self.cpp_info.libs = ["wavelet_buffer", "sf_compressor"]
        self.cpp_info.set_property("cmake_file_name", "wavelet_buffer")
        self.cpp_info.set_property("cmake_target_name", "wavelet_buffer::wavelet_buffer")
        
        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
        self.cpp_info.filenames["cmake_find_package"] = "wavelet_buffer"
        self.cpp_info.filenames["cmake_find_package_multi"] = "wavelet_buffer"
        self.cpp_info.names["cmake_find_package"] = "wavelet_buffer"
        self.cpp_info.names["cmake_find_package_multi"] = "wavelet_buffer"
```

Reply:
Since all works by default why we need explicit settings?

Reply:
Because it's better to be explicit and clearly reflect these are from upstream and nor generic Conan targets :)